File: Hex.py
"""
Cindy Zheng
Cybersecurity
XORstrings Assignment
"""

import sys
mode = sys.argv[1]
keyfile = sys.argv[2]
inpfile = sys.argv[3]
key = open(keyfile).read()[:-1] #removes the mandatory \n at the end of the file to support one line messages.
inp = open(inpfile).read()[:-1] #removes the mandatory \n at the end of the file to support one line messages.
debug = False

if(debug):
    print("mode:"+mode)
    print("key: "+key)
    print("inp: "+inp)
    
klen = len(key)
if (klen < len(inp)):
    for i in range(len(inp)-klen):
        key = key + key[i % klen]
        
    
def humanf(key, inp):
    outp = ""
    for i in range(len(inp)):
        outp = outp + chr((ord(key[i])^(ord(inp[i]))))
    return outp
    
def numOutf(key, inp):
    outp = ""
    for i in range(len(inp)):
        outp = outp + hex((ord(key[i])^(ord(inp[i]))))[2:] + " "
    return outp

if (mode=="human"):
    print(humanf(key,inp))
else:
    print(numOutf(key,inp))

File: Hex.java
/**************************************************************************    
Cindy Zheng                                                                    
Cybersecurity                                                                  
XOR Strings                                                                   
                                                                               
Given three strings mode, text, and key: it will produce an output string by x\
r'ing corresponding values together.                                           
 *************************************************************************/
import java.nio.file.*;
public class Hex{

    public static void main(String[]args){
	String mode;
	String keyfile;
	String inpfile;
	String key;
	String inp;
	boolean debug = true;
	try{
	    mode = args[0];
	    keyfile = args[1];
	    inpfile = args[2];
	    key = Files.readString(Path.of(keyfile));
	    key = key.substring(0, key.length());
	    inp = Files.readString(Path.of(inpfile));
	    inp = inp.substring(0, inp.length());
	    if(debug){
		System.out.println("mode:"+mode);
		System.out.println("key: "+key);
		System.out.println("inp: "+inp);
	    }
	}catch(Exception e){
	    e.printStackTrace();
	    System.exit(1);
	}
       
    }
}
